asset-transfer-events/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,210 @@
+        }
+    } catch (error: unknown) {
+        if (!(error instanceof GatewayError) || error.code !== grpc.status.CANCELLED) {
+            throw error;
+        }
+    }
+}
+
+/**
+ * Submit a transaction asynchronously to create a new asset.
+ */
+async function createAsset(contract: Contract): Promise<bigint> {
+    console.log(`\n --> Submit Transaction: CreateAsset, creates ${assetId} owned by Tom with appraised value 100`);
+
+    const result = await contract.submitAsync(
+        'CreateAsset',
+        {
+            arguments: [
+                assetId,
+                'yellow',
+                '5',
+                'Tom',
+                '100',]
+        });
+
+    const status = await result.getStatus();
+    if (!status.successful) {
+        throw new Error(`failed to commit transaction ${status.transactionId} with status code ${status.code}`);
+    }
+
+    console.log('\n*** CreateAsset committed successfully\n')
+
+    return status.blockNumber;
+
+}
+/**
+ * Submit transaction synchronously, to updateAsset the asset appraised value.
+ */
+async function updateAsset(contract: Contract): Promise<void> {
+    console.log(`\n--> Submit transaction: UpdateAsset, ${assetId} update appraised value to 200`);
+
+    await contract.submitTransaction('UpdateAsset',
+        assetId, 'yellow',
+        '5',
+        'Tom',
+        '200',
+    );
+
+    console.log('\n*** UpdateAsset committed successfully\n')
+
+}
+
+/**
+ * Submit transaction synchronously, to transfer the asset.
+ */
+async function transferAsset(contract: Contract): Promise<void> {
+    console.log(`\n--> Submit transaction: TransferAsset, ${assetId} to Saptha\n`);
+
+    await contract.submitTransaction('TransferAsset',
+        assetId, 'Saptha',
+    );
+
+    console.log('\n*** TransferAsset committed successfully\n')
+}
+
+/**
+ * Submit a transaction synchronously to delete an asset by ID.
+ */
+async function deleteAssetByID(contract:Contract): Promise<void>{
+    console.log(`\n--> Submit transaction: DeleteAsset ${assetId}`);
+
+    await contract.submitTransaction(
+        'DeleteAsset',
+        assetId
+    );
+
+    console.log('\n*** DeleteAsset committed successfully\n')
+
+}
+
+
+
+/**
+  * Replay all the events from the start block.
+  */
+async function replayChaincodeEvents(network:Network,startBlock:bigint):Promise<void>{
+
+    const events = await network.getChaincodeEvents(chaincodeName, {
+        startBlock
+    });
+    try {
+        for await (const event of events) {
+            const payload = utf8Decoder.decode(event.payload);
+            console.log(`<-- Chaincode event replayed: ${event.eventName}, payload: ${payload}`);
+            if(event.eventName === 'DeleteAsset'){
+                break
+            }
+        }
+    } catch (error: unknown) {
+        if (!(error instanceof GatewayError) || error.code !== grpc.status.CANCELLED) {
+            throw error;
+        }
```

Comment:
We don't expect the event stream to be closed until after we finish reading and break out of the loop (above), so should probably not be catching any errors here. Just remove the catch.